src/pip/_internal/operations/install/wheel.py

```diff
@@ -766,7 +766,7 @@ def _generate_file(path, **kwargs):
     # Record the REQUESTED file
     if requested:
         requested_path = os.path.join(dest_info_dir, 'REQUESTED')
-        with open(requested_path, "w"):
+        with open(requested_path, "wb"):
```

Comment:
As this simply opens then closes (creating an empty file), this makes no real difference. I'm inclined to prefer not to use "b" because that implies the file is binary, and it isn't. But I don't really care much.

`pathlib.touch` might be better, now we don't have to worry about Python 2 compatibility.

Reply:
I prefer binary mode because creating TextIOWrapper is waste of time/memory and using locale-specific encoding is root of bugs.
But I respect your preference.

Reply:
Actually, I went looking to see what encoding the `REQUESTED` file should use. https://packaging.python.org/specifications/recording-installed-packages/ says it's tool-specific, and refers to [PEP 376](https://www.python.org/dev/peps/pep-0376/#requested) which says it can be empty or contain a comment. So the content is essentially irrelevant, and binary mode therefore makes sense.

Let's use binary mode, and add a comment saying "Create an empty file (see PEP 376)", just to be clear that the encoding isn't relevant.